src/components/uikit/BottomSheet.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { useCallback, useRef, ReactNode } from 'react';
+import { useWindowDimensions } from 'react-native';
+import RNBottomSheet, {
+  BottomSheetProps as RNBottomSheetProps,
+} from '@gorhom/bottom-sheet';
+
+import { styled, useTheme } from '~styles';
+
+interface BottomSheetProps {
+  initialIndex: number;
+  snapPoints: string[]; // e.g. ['25%', '50%']
+  children: ReactNode;
+  onSheetChanges?: (index: number) => void;
+  keyboardBehavior?: RNBottomSheetProps['keyboardBehavior'];
+}
+
+export function BottomSheet({
+  initialIndex,
+  snapPoints,
+  children,
+  onSheetChanges,
+  keyboardBehavior = 'interactive',
+}: BottomSheetProps) {
+  const theme = useTheme();
+  const bottomSheetRef = useRef<RNBottomSheet>(null);
```

Comment:
This internal ref is not used for anything and it would be better if the bottom sheet ref would come from the parent so that the user of this component has access to the ref. So you need to wrap this UI kit component with `forwardRef`.